jdbc/src/java/org/apache/hive/jdbc/ZooKeeperHiveClientHelper.java

```diff
@@ -178,12 +179,13 @@ private static void updateParamsWithZKServerNode(JdbcConnectionParams connParams
     // it must be the server uri added by an older version HS2
     Matcher matcher = kvPattern.matcher(dataStr);
     if ((dataStr != null) && (!matcher.find())) {
-      String[] split = dataStr.split(":");
-      if (split.length != 2) {
+      try {
+        IPStackUtils.HostPort hostPort = IPStackUtils.splitHostPort(dataStr);
+        connParams.setHost(hostPort.getHostname());
+        connParams.setPort(hostPort.getPort());
+      } catch (Exception e) {
         throw new ZooKeeperHiveClientException("Unable to parse HiveServer2 URI from ZooKeeper data: " + dataStr);
       }
-      connParams.setHost(split[0]);
-      connParams.setPort(Integer.parseInt(split[1]));
```

Comment:
I think there is already a Util if the URL thing with // doesn't work, can you check once
```
HostAndPort hostAndPort = HostAndPort.fromString(instanceUri);
String hostname = hostAndPort.getHost();
int port = hostAndPort.getPort();
```

Reply:
This method also doesn't work if hostname is IPv6 and not enclosed in square brackets and what we are parsing is not a complete URI..